agreement/vote.go

```diff
@@ -182,6 +182,44 @@ func makeVote(rv rawVote, voting crypto.OneTimeSigner, selection *crypto.VRFSecr
 	return unauthenticatedVote{R: rv, Cred: cred, Sig: sig}, nil
 }
 
+// signVote sign a given raw vote and uses the provided credential.
+//
+// signVote returns an error it it fails.
+func signVote(rv rawVote, voting crypto.OneTimeSigner, l Ledger, cred committee.Credential) (vote, error) {
```

Comment:
This seems to duplicate a lot of code with `makeVote`. It would be nice to avoid this.

Reply:
I agree. I think that makeVote isn't needed at all ( eventually). But this change doesn't touch the non-proposal vote generation, which is why it's still there. ( i.e. the only other consumer for that in a non-test code is the asyncPseudonode::makeVotes.